tests/ui/traits/const-traits/match-non-const-eq.stock.stderr

```diff
@@ -1,12 +1,13 @@
-error[E0015]: cannot match on `str` in constant functions
-  --> $DIR/match-non-const-eq.rs:7:9
+error[E0658]: cannot call conditionally-const method `<str as PartialEq>::eq` in constant functions
```

Comment:
Nitpick: surprised that when rendering predicates we talk about `T: ~const PartialEq`, but in fully qualified paths we talk `<str as PartialEq>::eq` instead of something like `<str as ~const PartialEq>::eq`. We already mention "conditionally-const" in the text, but wonder how many diagnostics will have to account for that to properly convey that context.

Reply:
well, so, `<str as ~const PartialEq>::eq` is not a valid qpath. the method itself still is just `<str as PartialEq>::eq`, it just has an additional "const requirement" on it.

Reply:
like, `<str as ~const PartialEq>::eq` and `<str as PartialEq>::eq` being distinct suggests that they are different methods, but they are not. this is a consequence of the current implementation of conditional constness.

Reply:
this is somewhat related to the fact that we allow users to write `T: Trait<Assoc = i32>`, but not in method call position (i.e. `<T as Trait<Assoc = i32>>::method` is not valid). The constness of a trait is associated with a trait impl. Imagine that `T: ~const Trait` were written like `T: Trait<constness = ~const>`.